backend/services/videoPromptService.js

```diff
@@ -0,0 +1,174 @@
+
+    const ips = await PromptSpace.findOne({
+        _id: ipsId,
+        users: { $in: [userId] },
+        type: PROMPT_SPACE_TYPE.Video,
+    });
+
+    if (!ips) {
+        return {
+            error: new AppError("Video prompt space not found", 404),
+        };
+    }
+
+    const query = {};
+    if (cursor) {
+        query._id = { $lt: cursor };
+    }
+
+    const videoPrompts = await VideoPrompt.find({
+        promptSpaceId: ipsId,
+        ...query,
+    })
+        .limit(limit + 1)
+        .sort({ createdAt: -1 })
+        .exec();
+
+    const hasNext = videoPrompts.length > limit;
+
+    if (hasNext) {
+        videoPrompts.pop();
+    }
+
+    const newCursor = videoPrompts[videoPrompts.length - 1]?._id;
+
+    const videoPromptDtos = [];
+
+    for (const videoPrompt of videoPrompts) {
+        const videoPromptDto = VideoPromptDto.fromModel(videoPrompt);
+        await generatePresignedUrlForVideoPromptDto(videoPromptDto);
+        videoPromptDtos.push(videoPromptDto);
+    }
+
+    return {
+        data: videoPromptDtos,
+        pagination: new PaginationResponseDto({
+            cursor: newCursor,
+            limit,
+            hasNext,
+        }),
+    };
+};
+
+const createVideoPrompt = async ({ input, ipsId, userId }) => {
+    const cipInput = VideoPromptCreateDto.fromRequest(input);
+    const { message, samplingSteps, cfgScale, eta, fps } = cipInput;
+    const fullMessage = generateVideoPromptFullMessage(cipInput);
+
+    const ips = await PromptSpace.findOne({
+        _id: ipsId,
+        users: { $in: [userId] },
+        type: PROMPT_SPACE_TYPE.Video,
+    });
+
+    if (!ips) {
+        return {
+            error: new AppError("Video prompt space not found", 404),
+        };
+    }
+
+    try {
+        const { Client } = await import("@gradio/client");
+        const app = await Client.connect(
+            process.env.VIDEO_GENERATION_API_URL ||
+                "https://videocrafter-videocrafter.hf.space/"
+        );
+
+        const result = await app.predict(1, [
+            message,
+            samplingSteps || 16, // 默认采样步数
+            cfgScale || 15, // 默认 CFG scale
+            eta || 1, // 默认 ETA 值
+            fps || 5, // 默认帧速率
+        ]);
+
+        if (!result || !result.data || !result.data[0] || !result.data[0][0]) {
+            throw new AppError("Video generation failed", 500);
+        }
+
+        const video_id = result.data[0][0].name;
+        const video_url = `${process.env.VIDEO_GENERATION_API_BASE_URL || "https://videocrafter-videocrafter.hf.space"}/file=${video_id}`;
+
+        const extension = "mp4"; // 假设生成的视频格式为 mp4
+
+        // 下载视频并上传到 S3
+        const videoResponse = await axios.get(video_url, {
+            responseType: "arraybuffer",
+        });
+        const responseVideoBuffer = Buffer.from(videoResponse.data, "base64");
+
+        const newVideoPrompt = new VideoPrompt({
+            promptSpaceId: ips._id,
+            input: {
```

Comment:
filters: {
            width: Number,
            height: Number,
            aspectRatio: String,
            duration: Number,
            quality: {
                type: string,
                enum: VIDEO_QUALITY_VALUES,
            },
        },
        image: VideoPromptInputImageSchema,
        soundTrack: VideoPromptInputSoundTrackSchema,
        message: String,
        fullMessage: String,
        model: String,

This is the current model for video prompt. eta, fps these are new need add. Does the api also support other parameters, width and height and more. The current model data is made based on UI design.

Reply:
The truth is that I'm a bit torn because eta, fps have been tested by Bagus when accessing the api and are written as constants (other variables would cause the interface to be very slow). Do we need to add these two inside input{}, because I don't think we need to pass these two values in after the frontend?

Reply:
ok, then lets not use the filters, just keep it there in models but never use. But need to add the other fields your adding here to the model schema. This refers to the "input: {...}" object

Reply:
> Ok if too slow, then lets not use the filters, just keep it there in models but not use for now (and we remove filter on front end just to include a message). But need to add the other fields your adding here to the model schema. This refers to the "input: {...}" object

Ok I will do it